ui/app/src/services/content.ts

```diff
@@ -562,6 +646,69 @@ export function insertComponent(
   );
 }
 
+export function insertInstance(
+  site: string,
+  modelId: string,
+  fieldId: string,
+  targetIndex: number,
+  instance: ContentInstance,
+  parentModelId: string = null,
+): Observable<any> {
+  return performMutation(
+    site,
+    modelId,
+    parentModelId,
+    doc => {
+      const qs = {
+        site,
+        path: modelId,
+        unlock: 'true',
+        fileName: getInnerHtml(doc.querySelector('file-name'))
+      };
+
+      const id = instance.craftercms.id;
+      const pathBase = `/site/components/${instance.craftercms.contentType.replace('/component/', '')}s/`.replace(/\/{1,}$/m, '');
+      const path = `${pathBase}/${id}.xml`;
+
+      const newItem = doc.createElement('item');
+
+      createElements(doc, newItem, {
+        '@attributes': {
+          // TODO: Hardcoded value. Fix.
+          datasource: 'sharedFeatures'
+        },
+        key: path,
+        value: instance.craftercms.label,
+        ...{
+          include: path,
+          disableFlattening: 'false'
+        }
+      });
+
+      let fieldNode = doc.querySelector(`:scope > ${fieldId}`);
```

Comment:
Need to update to use extractNode, etc. Refactor the code duplication with `insertComponent`.